unity/Assets/Maroon/reusablePrefabs/NewRuler/Scripts/Test_RulerDisplay.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Assets.Maroon.reusablePrefabs.NewRuler.Scripts;
+using Maroon.GlobalEntities;
+using Maroon.Physics.CoordinateSystem;
+using TMPro;
+using UnityEngine;
+using Maroon.GlobalEntities;
+
+public class Test_RulerDisplay : MonoBehaviour
+{
+    [SerializeField] private RulerLogic ruler;
+
+    [SerializeField] private TMP_Text DistanceText;
+
+    [SerializeField] private GameObject StartPositionX;
+    [SerializeField] private GameObject StartPositionY;
+    [SerializeField] private GameObject StartPositionZ;
+
+    [SerializeField] private GameObject EndPositionX;
```

Comment:
Please use the Maroon.UI.InputField type instead of GameObject
Please use also the UI Prefabs from the \Maroon\Assets\Maroon\reusableGui\Experiment\Resources folder